From ac2e8624a125ed482a7046eb73d109e51f25b19b Mon Sep 17 00:00:00 2001
From: dimkaps <150019158+dimkaps@users.noreply.github.com>
Date: Mon, 8 Dec 2025 12:04:05 +0000
Subject: [PATCH] Include AK0991X magnetometer into ADSP build for kalama (#4)

---
 ssc/chipset/kailua/por.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ssc/chipset/kailua/por.py b/ssc/chipset/kailua/por.py
index 3f31988d..b221c75e 100755
--- a/ssc/chipset/kailua/por.py
+++ b/ssc/chipset/kailua/por.py
@@ -234,6 +234,7 @@ def generate(env):
     env.AddUsesFlags(['SNS_ISLAND_INCLUDE_LPS22DF'])
     env.AddUsesFlags(['SNS_ISLAND_INCLUDE_TMD2755'])
     env.AddUsesFlags(['SNS_ISLAND_INCLUDE_SHT4X'])
+    env.AddUsesFlags(['SNS_ISLAND_INCLUDE_AK0991X'])
 
     # 2. Calibration
     env.AddUsesFlags(['SNS_ISLAND_INCLUDE_GYRO_CAL'])
-- 
2.41.0

